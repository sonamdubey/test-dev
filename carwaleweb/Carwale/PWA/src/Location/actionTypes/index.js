export const OPEN_POPUP = 'OPEN_POPUP';
export const CLOSE_POPUP = 'CLOSE_POPUP';
export const SHOW_CROSS_ICON = 'SHOW_CROSS_ICON';
export const HIDE_CROSS_ICON = 'HIDE_CROSS_ICON';
