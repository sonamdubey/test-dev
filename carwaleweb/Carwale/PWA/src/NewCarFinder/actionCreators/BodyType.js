import {
	REQUEST_BODY_TYPE,
	RECEIVE_BODY_TYPE,
	BODY_TYPE_SELECTION,
	SET_BODY_STORED_FILTERS
} from '../actionTypes'
import {
	getNCFFilterParams
} from '../../filterPlugin/selectors/FilterPluginSelectors'
import BodyTypeApi from '../apis/BodyType'

import {
	getCityId
} from '../../selectors/LocationSelectors'

import {
	getBudgetFilter
} from '../selectors/NCFSelectors'

export const requestBodyType = () => {
	return {
		type: REQUEST_BODY_TYPE
	}
}

export const setBodyTypeData = data => {
	return {
		type: RECEIVE_BODY_TYPE,
		data
	}
}

export const selectBodyType = id => {
	return {
		type: BODY_TYPE_SELECTION,
		id
	}
}

export const fetchBodyType = () => (dispatch, getState) => {
	dispatch(requestBodyType())
	let state = getState()
	const currentScreen = state.newCarFinder.filter.filterScreens.currentScreen
	const prevFilters = state.newCarFinder.filter.filterScreens.screenOrder.slice(0, currentScreen).reduce((acc, screen) => acc.concat([screen.name]), [])
	const options = getNCFFilterParams(state, prevFilters)
	if(state.newCarFinder.shortlistCars.removedModelIds != undefined){
		options.removedModelIds = state.newCarFinder.shortlistCars.removedModelIds
	}
	BodyTypeApi
		.get(options)
		.then(data => { setTimeout(() => dispatch(setBodyTypeData(data)), 300) })
		.catch( error => {
			//TODO: how should error affect the state ?
			console.log(error)
		})
}

export const setBodyStoredFilters = storedFilters => {
	return {
		type: SET_BODY_STORED_FILTERS,
		storedFilters
	}
}
