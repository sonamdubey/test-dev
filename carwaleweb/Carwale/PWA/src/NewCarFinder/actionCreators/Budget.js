import {
	UPDATE_BUDGET_INPUT_VALUE,
	UPDATE_BUDGET_SLIDER_VALUE
} from '../actionTypes'

import {
	fireInteractiveTracking
} from '../../utils/Analytics'
const updateSliderValue = (values, min, max, userChange) => {
	return {
		type: UPDATE_BUDGET_SLIDER_VALUE,
		values,
		min,
		max,
		userChange
	}
}

const updateBudgetValue = (value) => {
	return {
		type: UPDATE_BUDGET_INPUT_VALUE,
		value
	}
}

const shouldSliderUpdate = (values, slider) => {
	let update = true

	if (slider.values[0] === values[0]) {
		update = false
	}

	return update
}

export const updateBudgetSlider = ({values, min, max, userChange}) => (dispatch, getState) => {
	if (shouldSliderUpdate(values, getState().newCarFinder.budget.slider)) {
		dispatch(updateSliderValue(values, min, max, userChange))
		dispatch(updateBudgetValue(values[0]))
	}
}
