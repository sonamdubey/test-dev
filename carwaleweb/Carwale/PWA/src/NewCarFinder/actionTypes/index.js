export const NEXT_FILTER = 'NEXT_FILTER'
export const SET_CURRENT_SCREEN = 'SET_CURRENT_SCREEN'
export const SET_ALL_FILTER_SCREEN = 'SET_ALL_FILTER_SCREEN'

export const REQUEST_CAR_FEATURE = 'REQUEST_CAR_FEATURE'
export const RECEIVE_CAR_FEATURE = 'RECEIVE_CAR_FEATURE'
export const CAR_FEATURE_SELECTION = 'CAR_FEATURE_SELECTION'

// budget
export const UPDATE_BUDGET_SLIDER_VALUE = 'UPDATE_BUDGET_SLIDER_VALUE'
export const UPDATE_BUDGET_INPUT_VALUE = 'UPDATE_BUDGET_INPUT_VALUE'

// body type
export const REQUEST_BODY_TYPE = 'REQUEST_BODY_TYPE'
export const RECEIVE_BODY_TYPE = 'RECEIVE_BODY_TYPE'
export const BODY_TYPE_SELECTION = 'BODY_TYPE_SELECTION'
export const SET_BODY_STORED_FILTERS = 'SET_BODY_STORED_FILTERS'


// fuel type
export const REQUEST_FUEL_TYPE = 'REQUEST_FUEL_TYPE'
export const RECEIVE_FUEL_TYPE = 'RECEIVE_FUEL_TYPE'
export const FUEL_TYPE_SELECTION = 'FUEL_TYPE_SELECTION'
export const SET_FUEL_STORED_FILTERS = 'SET_FUEL_STORED_FILTERS'

// fuel type
export const REQUEST_MAKE = 'REQUEST_MAKE'
export const RECEIVE_MAKE = 'RECEIVE_MAKE'
export const MAKE_SELECTION = 'MAKE_SELECTION'
export const SET_MAKE_STORED_FILTERS = 'SET_MAKE_STORED_FILTERS'



// listing
// model list RECEIVE_MODEL_LIST
export const REQUEST_MODEL_LIST = 'REQUEST_MODEL_LIST'
export const RECEIVE_MODEL_LIST = 'RECEIVE_MODEL_LIST'
export const APPEND_MODEL_LIST = 'APPEND_MODEL_LIST'

// filter screen
export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'

// compare cars
export const ADD_MODEL_TO_COMPARE = 'ADD_MODEL_TO_COMPARE'
export const REMOVE_MODEL_FROM_COMPARE = 'REMOVE_MODEL_FROM_COMPARE'
// header
export const MAKE_HEADER_STICKY = 'MAKE_HEADER_STICKY'
export const MAKE_HEADER_NORMAL = 'MAKE_HEADER_NORMAL'
