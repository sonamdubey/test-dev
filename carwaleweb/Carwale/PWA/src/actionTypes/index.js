// location: city
export const SELECT_CITY = "SELECT_CITY";
export const SELECT_AREA = "SELECT_AREA";
export const SET_LOCATION = "SET_LOCATION";
export const PREFILL_LOCATION = "PREFILL_LOCATION";

// toast
export const SET_TOAST = "SET_TOAST";
export const REMOVE_TOAST = "REMOVE_TOAST";

// header
export const SHOW_HEADER = "SHOW_HEADER";
export const HIDE_HEADER = "HIDE_HEADER";

// footer
export const SHOW_FOOTER = "SHOW_FOOTER";
export const HIDE_FOOTER = "HIDE_FOOTER";

// navigation drawer
export const OPEN_NAV_DRAWER = "OPEN_NAV_DRAWER";
export const CLOSE_NAV_DRAWER = "CLOSE_NAV_DRAWER";

// fullscreen
export const HIDE_HEADER_FOOTER = "HIDE_HEADER_FOOTER";
export const SHOW_HEADER_FOOTER = "SHOW_HEADER_FOOTER";

//page
export const SET_PAGE = "SET_PAGE";

// autoDetect location
export const AUTODETECT_LOCATION = "AUTODETECT_LOCATION"

export const SELECT_LATLONG = "SELECT_LATLONG";
export const FETCHING_LOCATION = "FETCHING_LOCATION"
export const SELECT_AREA_CITIES = "SELECT_AREA_CITIES"
