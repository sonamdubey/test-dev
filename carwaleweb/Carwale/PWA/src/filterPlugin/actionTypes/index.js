export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'
export const SET_FILTERS = 'SET_FILTERS'
