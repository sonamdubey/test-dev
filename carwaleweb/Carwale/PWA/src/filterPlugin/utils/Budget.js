export const formatBudgetTooltipValue = (value) => {
	let budgetValue

	if (value > 9999999) {
		budgetValue = parseFloat((value / 10000000).toFixed(2)) + '+ Cr'
	}
	else if (value <= 200000) {
		budgetValue = 'Upto 2 lakh'
	}
	else {
		budgetValue = parseFloat((value / 100000).toFixed(2)) + ' lakh'
	}

	return budgetValue
}
