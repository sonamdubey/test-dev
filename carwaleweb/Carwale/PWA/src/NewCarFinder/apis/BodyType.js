
import { SUGGESTED_BODYTYPE_ENDPOINT } from '../constants'
import { serialzeObjectToQueryString } from '../../utils/Common'

/**
 * This returns a promise to fetch
 * suggested bodytype in a given budget,city
 * @return {Promise<>}
 */
const get = (options) => {
    return fetch(SUGGESTED_BODYTYPE_ENDPOINT+"?"+serialzeObjectToQueryString(options))
    .then(response => {
        if (!response.ok) { throw response }
        return response.json()
    })
}
const BodyTypeApi = {
    get
}
export default BodyTypeApi
