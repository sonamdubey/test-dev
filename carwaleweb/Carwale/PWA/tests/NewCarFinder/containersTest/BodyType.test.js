import React from 'react'
import { Redirect, MemoryRouter } from 'react-router-dom'

import BodyType from '../../../src/NewCarFinder/containers/BodyType'
import { SET_TOAST } from '../../../src/actionTypes'

import {
	SET_CURRENT_SCREEN,
	REQUEST_BODY_TYPE,
	RECEIVE_BODY_TYPE,
	BODY_TYPE_SELECTION
} from '../../../src/NewCarFinder/actionTypes'

import { formatBudgetTooltipValue } from '../../../src/NewCarFinder/utils/Budget'

const bodyTypeApiResponse = [
    { "id": 6, "name": "SUV/MUV", "carCount": 8, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv.svg", "isSelected": true, "isRecommended": true },
    { "id": 3, "name": "Hatchback", "carCount": 7, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback.svg", "isSelected": true, "isRecommended": true },
    { "id": 1, "name": "Sedan", "carCount": 5, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan.svg", "isSelected": false, "isRecommended": false },
    { "id": 5, "name": "Convertible", "carCount": 1, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible.svg", "isSelected": false, "isRecommended": false },
    { "id": 2, "name": "Coupe", "carCount": 1, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe.svg", "isSelected": false, "isRecommended": false },
    { "id": 10, "name": "Compact Sedan", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan.svg", "isSelected": false, "isRecommended": false },
    { "id": 4, "name": "Minivan/Van", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van.svg", "isSelected": false, "isRecommended": false },
    { "id": 8, "name": "Station Wagon", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon.svg", "isSelected": false, "isRecommended": false },
    { "id": 7, "name": "Truck", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck.svg", "isSelected": false, "isRecommended": false }
]


// Mock body type api
jest.mock('../../../src/NewCarFinder/apis/BodyType', () => {
    const bodyTypeApiResponse = [
        { "id": 6, "name": "SUV/MUV", "carCount": 8, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv.svg", "isSelected": true, "isRecommended": true },
        { "id": 3, "name": "Hatchback", "carCount": 7, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback.svg", "isSelected": true, "isRecommended": true },
        { "id": 1, "name": "Sedan", "carCount": 5, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan.svg", "isSelected": false, "isRecommended": false },
        { "id": 5, "name": "Convertible", "carCount": 1, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible.svg", "isSelected": false, "isRecommended": false },
        { "id": 2, "name": "Coupe", "carCount": 1, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe.svg", "isSelected": false, "isRecommended": false },
        { "id": 10, "name": "Compact Sedan", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan.svg", "isSelected": false, "isRecommended": false },
        { "id": 4, "name": "Minivan/Van", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van.svg", "isSelected": false, "isRecommended": false },
        { "id": 8, "name": "Station Wagon", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon.svg", "isSelected": false, "isRecommended": false },
        { "id": 7, "name": "Truck", "carCount": 0, "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.", "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck_clr.svg", "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck.svg", "isSelected": false, "isRecommended": false }
    ]
    return {
        get: jest.fn((cityId, budget) => Promise.resolve(bodyTypeApiResponse))
    }
})

//Mock Ripple js
jest.mock('../../../src/utils/Ripple')

describe("Tests for BodyType container component", () => {

    let props
    let store
    let enzymeWrapper
    let state
    const Wrapper = () => {
        if (!enzymeWrapper) {
            store = mockStore(state)
            enzymeWrapper = shallowWithStore(<BodyType {...props} />, store).dive()
        }
        return enzymeWrapper
    }

    beforeEach(() => {
        props = undefined
        state = getMockInitialStore()
        enzymeWrapper = undefined
        store = undefined
    })

    describe("when no body type data is available", () => {
        it("should match snapshot", () => {
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when component is mounted", () => {
        it("should dispatch appropriate actions", (done) => {
            /**
             * documentation for done and use case : https://jasmine.github.io/2.5/introduction.html#section-58
             * and https://stackoverflow.com/questions/47275706/async-function-not-behaving-as-i-expect-with-jest
             */
            const wrapper = Wrapper()
            const expectedActions = store.getActions()
            expect(expectedActions).toHaveLength(2)

            expect(expectedActions).toContainEqual({
                type: SET_CURRENT_SCREEN,
                screenId: 1
            })

            expect(expectedActions).toContainEqual({
                type: REQUEST_BODY_TYPE
            })

            //http://www.bambielli.com/til/2017-09-19-process-nexttick/
            process.nextTick(() => {
                setTimeout(() => {
                    try {
                        expect(store.getActions()).toHaveLength(3)
                        expect(store.getActions()).toContainEqual({
                            type: RECEIVE_BODY_TYPE,
                            data: bodyTypeApiResponse
                        })
                    } catch (e) {
                        return done(e);
                    }
                    done();
                }, 300)
            });
        })
    })

    describe("when data is available", () => {
        it("should match snapshot", () => {
            state.newCarFinder.bodyType.data = bodyTypeApiResponse
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when bodyType with count > 0 is clicked", () => {
        it("should dispatch BODY_TYPE_SELECTION action", () => {
            state.newCarFinder.bodyType.data = bodyTypeApiResponse
            const wrapper = Wrapper()
            let activeItems = bodyTypeApiResponse.filter(x => x.carCount > 0)
            if (activeItems.length) {
                const firstItem = wrapper.find('.body-type-list__item').findWhere(x => x.key() == activeItems[0].id)
                firstItem.simulate('click')
                const expectedActions = store.getActions()
                expect(expectedActions).toContainEqual({
                    type: BODY_TYPE_SELECTION,
                    id: parseInt(firstItem.key())
                })
            }
        })
    })

    describe("when bodyType with count == 0 is clicked", () => {
        it("should dispatch BODY_TYPE_SELECTION action", () => {
            state.newCarFinder.bodyType.data = bodyTypeApiResponse
            const wrapper = Wrapper()
            let disabledItems = bodyTypeApiResponse.filter(x => x.carCount == 0)
            if (disabledItems.length) {
                const firstItem = wrapper.find('.body-type-list__item').findWhere(x => x.key() == disabledItems[0].id)
                firstItem.simulate('click')
                const expectedActions = store.getActions().filter(x => x.type == SET_TOAST)
                expect(expectedActions.length).toBe(1)
                expect(expectedActions[0].message).toBe(`There is no ${disabledItems[0].name} available for ${formatBudgetTooltipValue(wrapper.instance().props.budget)} in ${wrapper.instance().props.cityName}.`)
            }
        })
    })

})
