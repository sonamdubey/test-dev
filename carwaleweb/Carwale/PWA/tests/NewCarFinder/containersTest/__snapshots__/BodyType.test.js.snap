// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for BodyType container component when data is available should match snapshot 1`] = `
<div
  className="body-type-screen"
>
  <div
    className="screen__head"
  >
    <div
      className="screen-head__image"
    />
    <p
      className="screen-head__title"
    >
      Select your preferred body type
    </p>
    <p
      className="screen-head__subtitle"
    >
      Recommended Body Types pre-selected
    </p>
    <ul
      className="body-type__list list__recommended-body"
    >
      <li
        className="body-type-list__item list-recommended-body__item active"
        key="6"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType=""
          item={
            Object {
              "carCount": 8,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg",
              "id": 6,
              "isRecommended": true,
              "isSelected": true,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv.svg",
              "name": "SUV/MUV",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-recommended-body__item active"
        key="3"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType=""
          item={
            Object {
              "carCount": 7,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback_clr.svg",
              "id": 3,
              "isRecommended": true,
              "isSelected": true,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback.svg",
              "name": "Hatchback",
            }
          }
          showCount={true}
        />
      </li>
    </ul>
  </div>
  <div
    className="screen__body"
  >
    <p
      className="screen-body__title"
    >
      Other body types for selection
    </p>
    <ul
      className="body-type__list list__other-body"
    >
      <li
        className="body-type-list__item list-other-body__item "
        key="1"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 5,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan_clr.svg",
              "id": 1,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan.svg",
              "name": "Sedan",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-other-body__item "
        key="5"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 1,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible_clr.svg",
              "id": 5,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible.svg",
              "name": "Convertible",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-other-body__item "
        key="2"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 1,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe_clr.svg",
              "id": 2,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe.svg",
              "name": "Coupe",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-other-body__item disable"
        key="10"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 0,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan_clr.svg",
              "id": 10,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan.svg",
              "name": "Compact Sedan",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-other-body__item disable"
        key="4"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 0,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van_clr.svg",
              "id": 4,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van.svg",
              "name": "Minivan/Van",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-other-body__item disable"
        key="8"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 0,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon_clr.svg",
              "id": 8,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon.svg",
              "name": "Station Wagon",
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="body-type-list__item list-other-body__item disable"
        key="7"
        onClick={[Function]}
      >
        <BodyTypeItem
          iconType="line"
          item={
            Object {
              "carCount": 0,
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
              "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck_clr.svg",
              "id": 7,
              "isRecommended": false,
              "isSelected": false,
              "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck.svg",
              "name": "Truck",
            }
          }
          showCount={true}
        />
      </li>
    </ul>
  </div>
  <BodyTypeKnowMore
    bodyTypeData={
      Array [
        Object {
          "carCount": 8,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg",
          "id": 6,
          "isRecommended": true,
          "isSelected": true,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv.svg",
          "name": "SUV/MUV",
        },
        Object {
          "carCount": 7,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback_clr.svg",
          "id": 3,
          "isRecommended": true,
          "isSelected": true,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/hatchback.svg",
          "name": "Hatchback",
        },
        Object {
          "carCount": 5,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan_clr.svg",
          "id": 1,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/sedan.svg",
          "name": "Sedan",
        },
        Object {
          "carCount": 1,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible_clr.svg",
          "id": 5,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/convertible.svg",
          "name": "Convertible",
        },
        Object {
          "carCount": 1,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe_clr.svg",
          "id": 2,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/coupe.svg",
          "name": "Coupe",
        },
        Object {
          "carCount": 0,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan_clr.svg",
          "id": 10,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/compactsedan.svg",
          "name": "Compact Sedan",
        },
        Object {
          "carCount": 0,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van_clr.svg",
          "id": 4,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/van.svg",
          "name": "Minivan/Van",
        },
        Object {
          "carCount": 0,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon_clr.svg",
          "id": 8,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/wagon.svg",
          "name": "Station Wagon",
        },
        Object {
          "carCount": 0,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
          "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck_clr.svg",
          "id": 7,
          "isRecommended": false,
          "isSelected": false,
          "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/truck.svg",
          "name": "Truck",
        },
      ]
    }
    clickSource={
      Object {
        "BODY_TYPE_SCREEN": 1,
        "KNOW_MORE_BODY": 3,
        "KNOW_MORE_HEADER": 2,
      }
    }
    getItemStatus={[Function]}
    handleBodyTypeClick={[Function]}
  />
</div>
`;

exports[`Tests for BodyType container component when no body type data is available should match snapshot 1`] = `
<div
  className="body-type-screen"
>
  <div
    className="screen__head"
  >
    <div
      className="screen-head__image"
    />
    <p
      className="screen-head__title"
    >
      Select your preferred body type
    </p>
    <p
      className="screen-head__subtitle"
    >
      Recommended Body Types pre-selected
    </p>
    <ul
      className="body-type__list list__recommended-body"
    />
  </div>
  <div
    className="screen__body"
  >
    <p
      className="screen-body__title"
    >
      Other body types for selection
    </p>
    <ul
      className="body-type__list list__other-body"
    />
  </div>
</div>
`;
